bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairSerialBridgeHandler.java

```diff
@@ -12,133 +12,125 @@
+import org.openhab.core.io.transport.serial.SerialPort;
+import org.openhab.core.io.transport.serial.SerialPortIdentifier;
+import org.openhab.core.io.transport.serial.SerialPortManager;
+import org.openhab.core.io.transport.serial.UnsupportedCommOperationException;
 import org.openhab.core.thing.Bridge;
 import org.openhab.core.thing.ThingStatus;
 import org.openhab.core.thing.ThingStatusDetail;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import gnu.io.CommPort;
-import gnu.io.CommPortIdentifier;
-import gnu.io.NoSuchPortException;
-import gnu.io.PortInUseException;
-import gnu.io.SerialPort;
-import gnu.io.UnsupportedCommOperationException;
-
 /**
  * Handler for the IPBridge. Implements the connect and disconnect abstract methods of {@link PentairBaseBridgeHandler}
  *
  * @author Jeff James - initial contribution
  *
  */
+@NonNullByDefault
 public class PentairSerialBridgeHandler extends PentairBaseBridgeHandler {
     private final Logger logger = LoggerFactory.getLogger(PentairSerialBridgeHandler.class);
 
+    public PentairSerialBridgeConfig config = new PentairSerialBridgeConfig();
     /** SerialPort object representing the port where the RS485 adapter is connected */
-    SerialPort port;
+    private final SerialPortManager serialPortManager;
+    private Optional<SerialPort> port = Optional.empty();
+    private @Nullable SerialPortIdentifier portIdentifier;
 
-    public PentairSerialBridgeHandler(Bridge bridge) {
+    public PentairSerialBridgeHandler(Bridge bridge, SerialPortManager serialPortManager) {
         super(bridge);
+        this.serialPortManager = serialPortManager;
     }
 
     @Override
-    protected synchronized void connect() {
-        PentairSerialBridgeConfig configuration = getConfigAs(PentairSerialBridgeConfig.class);
+    protected synchronized int connect() {
+        config = getConfigAs(PentairSerialBridgeConfig.class);
+
+        if (config.serialPort.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "no serial port configured");
+            return -1;
+        }
+
+        this.id = config.id;
+        logger.debug("Serial port id: {}", id);
+        this.discovery = config.discovery;
+
+        portIdentifier = serialPortManager.getIdentifier(config.serialPort);
+        if (portIdentifier == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configured serial port does not exist");
+            return -1;
+        }
 
         try {
-            CommPortIdentifier ci = CommPortIdentifier.getPortIdentifier(configuration.serialPort);
-            CommPort cp = ci.open("openhabpentairbridge", 10000);
-            if (cp == null) {
-                throw new IllegalStateException("cannot open serial port!");
+            logger.debug("connect port: {}", config.serialPort);
+
+            Objects.requireNonNull(portIdentifier, "portIdentifier is null");
```

Comment:
How is it possible to be null here? Is this check required?

Reply:
You are correct that this won't be null here, but I inserted this to eliminate a null warning.  Is there a better method to do this - or should I just ignore the warning?